src/mpid/ch3/channels/nemesis/include/mpid_nem_datatypes.h

```diff
@@ -146,10 +146,7 @@ typedef struct MPID_nem_pkt_header
 typedef struct MPID_nem_pkt
 {
     MPID_nem_pkt_header_t header;
-    union {
-        char payload[MPID_NEM_MPICH_DATA_LEN];
-        double dummy; /* align paylod to double */
-    } p;
+    char payload[] MPL_ATTR_ALIGNED(8);   /* C99 flexible array member with 8-byte alignment */
```

Comment:
Similar question here about assertion failure. Aligning the flexible array member seems a little questionable, but if compilers are able to handle it then it may be OK.

Reply:
To my understanding, there are two aspects as far as compiler concerned. The first one is address calculation, and the flexible array member is treated just as a regular array member. The second aspect is to calculate total size for memory allocation purposes. For that compiler is treating flexible array member as if it is not there.

Alignment is just for address calculation, so I think should be no problem for compilers.

In the end, if the assertion succeeds, then we are good for that compiler and platform. If the assertion fails on some compiler, I think we can try to figure out better solutions then. :)